src/node/history.h

```diff
@@ -676,67 +676,72 @@ namespace ccf
         return;
       }
 
-      auto txid = store.next_txid();
-      auto commit_txid = consensus->get_committed_txid();
-
-      LOG_DEBUG_FMT(
-        "Signed at {} in view: {} commit was: {}.{}",
-        txid.version,
-        txid.term,
-        commit_txid.first,
-        commit_txid.second);
+      auto signable_txid = consensus->get_signable_txid();
 
-      store.commit(
-        txid,
-        [txid, commit_txid, this]() {
-          kv::Tx sig(txid.version);
-          auto sig_view = sig.get_view(signatures);
-          crypto::Sha256Hash root = replicated_state_tree.get_root();
-
-          Nonce hashed_nonce;
-          auto consensus = store.get_consensus();
-          if (consensus != nullptr && consensus->type() == ConsensusType::BFT)
-          {
-            auto progress_tracker = store.get_progress_tracker();
-            CCF_ASSERT(
-              progress_tracker != nullptr, "progress_tracker is not set");
-            auto r = progress_tracker->record_primary(
-              txid.term, txid.version, id, root, hashed_nonce);
-            if (r != kv::TxHistory::Result::OK)
+      if (signable_txid.has_value())
```

Comment:
Minor: This would be match the pattern above (and a smaller diff) with a comment, negative check and early return. ie

```
// Signatures are only emitted when the consensus knows what is safely signable
if (!signable_txid.has_value())
{
  return;
}
```

Reply:
Good point